### ISSUE TITLE
✨ Repeatable random numbers

### ISSUE THREAD

**Issue by cbersch** [AutoFixture/AutoFixture#1406]
### Description
For some of my tests I need repeatable random numbers, e.g. to test serialization of data objects, for which I must compare the resulting string with a reference string, which isn't currently possible.

For this, I would need to configure the seed of the random number generators used in the `RegularExpressionGenerator` and `RandomNumericSequenceGenerator`.

I could give it a try, but I'm not sure how to implement this in accordance with the library's design principles.

### Suggestion

Maybe one could create a `RandomProvider` which is injected to the above two generators and provides a `Random` instance.

However, an instance of this `RandomProvider` must be delivered through several constructors to the `DefaultPrimitiveBuilders` constructor. 
I think passing this option later at runtime isn't feasible and not reasonable.

```cs

public class RandomProvider
{
    public Random Random { get; } = new Random();
    public void SetSeed(int seed) => Random = new Random(seed);
}

public interface IFixture
{
    void SetRandomSeed(int seed);
}

public class Fixture
{
    private readonly RandomProvider random = new RandomProvider();

    public Fixture() 
        : this(new RandomProvider()) { }

    public Fixture(RandomProvider random)
        : this(new DefaultEngineParts(random))
    {
        this.random = random ?? throw new ArgumentNullException(nameof(random));
    }

    public void SetRandomSeed(int seed) => random.SetSeed(seed);
}

// usage
IFixture fixture = new Fixture();
fixture.SetRandomSeed(27);
```

**Comment by jzabroski** [comment]
Adding a random number generator to the root Fixture object is not going to guarantee repeatable random numbers, as that may depend on the order in which fixture customizations are registered, which may or may not be guaranteed for multiple reasons (e.g., adding a new customization with a lexicographic name that gets embedded in the ECMA-335 assembly metadata earlier than code contributed earlier chronologically). A weaker guarantee would be to simply use the same seed.

Here is a simple example from my code base:

```c#
    public class SICDataInitializer : DataInitializerBase<SIC>
    {
        public SICDataInitializer(IDbContextLocator dbContextLocator) :
            base(dbContextLocator)
        { }

        public override void Customize(IFixture fixture)
        {
            base.Customize(fixture);
            fixture.Customize<SIC>(x => x
                .Do(p => p.Id = CreateUniqueString(fixture,3))
                .Do(p => p.Group = fixture.Create<SICIndustryGroup>())
                .Without(p => p.Id)
                .Without(p => p.Group)
            );
        }
    }
```
CreateUniqueString is defined in DataInitializerBase<T>, but the point is that in the two calls to `Do`, they recursively reference the fixture itself, so the exact materialization of values _over the object graph_ depends on a total ordering of calls to `IFixture.Create<T>`

**Comment by cbersch** [comment]
Correct, that doesn't guarantee repeatable random numbers, but at least then it's up to me to make that sure. Currently, I have now possibility to do that.

In my case I simply wanted to always generate the same float array for a test method.
